matter_server/server/device_controller.py

```diff
@@ -186,16 +186,22 @@ async def commission_with_code(
         LOGGER.info(
             "Starting Matter commissioning with code using Node ID %s.", node_id
         )
-        success = await self._call_sdk(
-            self.chip_controller.CommissionWithCode,
-            setupPayload=code,
-            nodeid=node_id,
-            networkOnly=network_only,
-        )
-        if not success:
-            raise NodeCommissionFailed(
-                f"Commission with code failed for node {node_id}"
+        retries = 3
+        while retries:
+            success = await self._call_sdk(
+                self.chip_controller.CommissionWithCode,
+                setupPayload=code,
+                nodeid=node_id,
+                networkOnly=network_only,
             )
+            if success:
+                break
+            if not success and not retries:
+                raise NodeCommissionFailed(
+                    f"Commission with code failed for node {node_id}"
+                )
+            await asyncio.sleep(5)
+            retries -= 1
```

Comment:
Can you add a message here that we are retrying? It will be helpful to understand logs better when we know that multiple commissioning attempts are being made for a particular device.

Reply:
I have adjusted the logging, now we will log the attempt too